crates/rattler_repodata_gateway/src/sparse/mod.rs

```diff
@@ -165,14 +165,14 @@ impl SparseRepoData {
     /// This works by iterating over all elements in the `packages` and
     /// `conda_packages` fields of the repodata and returning the unique
     /// package names.
-    pub fn package_names(&self) -> impl Iterator<Item = &'_ str> + '_ {
+    pub fn package_names(&self) -> impl Iterator<Item = &'_ str> {
         let repo_data = self.inner.borrow_repo_data();
-        repo_data
-            .packages
+        let tar_baz2_packages = repo_data.packages.iter().map(|(name, _)| name.package);
+        let conda_packages = repo_data
+            .conda_packages
             .iter()
-            .chain(repo_data.conda_packages.iter())
-            .map(|(name, _)| name.package)
-            .dedup()
+            .map(|(name, _)| name.package);
+        tar_baz2_packages.merge(conda_packages).dedup()
```

Comment:
Do you know if this prefers the `conda` package or the `tar.bz2`?

Reply:
So how it works is that we have two lists, one for tar.bz2 and one for .conda. Both lists are already sorted by package name. `merge` creates a new iterator that is also already sorted. Then `dedup` removes duplicate _repeated_ items.

Reply:
Oh and the names are just `foobar` in `foobar-0.1.0-blabla.conda`. So it doesnt matter which we take first.

Reply:
Previously the two sorted lists would be concattenated (or chained) and then deduped. So, you could have `[a,b,c,a,b]`, the result would not be a sorted list. We then called `dedup` on that list which for this example would not do anything because there are no _repeated_ duplicates. Now, instead of `chain` we use `merge` and you get this list `[a,a,b,b,c]`.

Reply:
I think ideally we'd have a preference for `.conda` packages and make this deterministic if possible?

Reply:
Or am I missing something and it doesn't matter for this particular function?

Reply:
Ok, looks like this function is only concerned with `package_names` so it doesn't matter where it comes from. Sorry for the noise.